0000-ldap-sync-tool.md

```diff
@@ -0,0 +1,139 @@
+- Feature Name: `mgr-ldapsync`: LDAP sync tool
+- Start Date: 2019, Oct 28
+- RFC PR: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+The `mgr-ldapsync` is a tool that performs synchronisation of the
+users between Uyuni server and LDAP.
+
+# Motivation
+[motivation]: #motivation
+
+Uyuni server can authenticate users from LDAP via PAM. For this, users
+needs to be added to Uyuni with the flag "PAM authentication" and thus
+their credentials will be verified via LDAP directory. Synchronisation
+tool should transparently synchronise users from certain groups and/or
+roles in LDAP and reflect that in Uyuni server, overriding local
+changes.
+
+Often LDAP configuration at infrastructures is unpredictable. This
```

Comment:
What kind of "unpredictability" are we talking about?

Reply:
There exists multiple schemas to configure users and groups in directory services.
Every Directory Service use there own standards.
And very often admins create additional schemas on top or use there own definitions.

Not sure if "unpredictable" is the right word for it.

Reply:
@mcalmer but those are still DN paths to those groups and users and those are still POSIX or Active Directory (lately almost compatible with POSIX) schemas.

Reply:
What word you would use instead?

Reply:
"As there are several ways and approaches to configure users and groups in directory services, this tool aims to work with most of them with a simple configuration setup." ?

Reply:
@thallma this is a poor explanation. :wink: What I wanted to say here is that there are few standard ways of managing groups/roles in LDAP. So this tool supposed to handle them all, having good 80% covered. 20% of those extraordinary setups I leave behind, without saying "No, never". But in those cases we would need a clear examples why current tool cannot work in their setup to find out what we can further help them as well.

Does it make more sense now?

Reply:
@paususe I've updated this.